### PR TITLE
KER-280 | gdpr api data fetch changes

### PR DESCRIPTION
**Pull request by nicobav**
- display only the user's comments
- don't display votes or voted comments

Refs KER-280

### CODE REVIEW

**Comment by terovirtanen** [comment]
**KERROKANTASI-API branch is deployed to platta: https://kerrokantasi-pr483.api.dev.hel.ninja** :rocket::rocket::rocket:

File: kerrokantasi/models.py [hunk 1]
```diff
@@ -26,11 +27,7 @@ class User(AbstractUser, SerializableMixin):
     # Properties for GDPR api serialization
     @property
     def sectioncomments(self):
-        return [s.serialize() for s in self.sectioncomment_created.everything().iterator()]
-
-    @property
-    def voted_sectioncomments(self):
-        return [s.serialize() for s in self.voted_democracy_sectioncomment.everything().iterator()]
+        return [s.serialize() for s in SectionComment.objects.everything().filter(created_by=self).iterator()]
```
Comment (charn) [inline kerrokantasi/models.py:30]: Not a huge thing, but I guess you can also do filtering in the everything method i.e.:
```suggestion
        return [s.serialize() for s in SectionComment.objects.everything(created_by=self).iterator()]
```

Comment (nicobav) [reply kerrokantasi/models.py]: fix'd

**Comment by terovirtanen** [comment]
**KERROKANTASI-API branch is deployed to platta: https://kerrokantasi-pr483.api.dev.hel.ninja** :rocket::rocket::rocket:

File: kerrokantasi/tests/gdpr/test_gdpr_query.py [hunk 2]
```diff
@@ -209,12 +209,9 @@ def _get_user_data(user: User) -> List[dict]:
...
         {
             "key": "SECTIONCOMMENTS",
-            "value": [_get_section_comment_data(comment) for comment in user.sectioncomment_created.everything()],
-        },
-        {
-            "key": "VOTED_SECTIONCOMMENTS",
             "value": [
-                _get_section_comment_data(comment) for comment in user.voted_democracy_sectioncomment.everything()
+                _get_section_comment_data(comment)
+                for comment in SectionComment.objects.everything().filter(created_by=user).iterator()
```
Comment (charn) [inline kerrokantasi/tests/gdpr/test_gdpr_query.py:214]:
```suggestion
                for comment in SectionComment.objects.everything(created_by=user).iterator()
```

File: kerrokantasi/tests/gdpr/test_gdpr_query.py [hunk 3]
```diff
@@ -266,11 +263,24 @@ def test_get_user_information_from_gdpr_api(user, geojson_feature):
...
     response = do_query(user, user.uuid)
 
     assert response.status_code == status.HTTP_200_OK
     _assert_user_data_in_response(response, user)
 
+    # Match the comment count
+    user_data = response.json()["children"]
+    section_comments = [child for child in user_data if child["key"] == "SECTIONCOMMENTS"][0]
+    all_comment_count = SectionComment.objects.everything().count()
+    user_comment_count = SectionComment.objects.everything().filter(created_by=user).count()
```
Comment (charn) [inline kerrokantasi/tests/gdpr/test_gdpr_query.py:279]:
```suggestion
    user_comment_count = SectionComment.objects.everything(created_by=user).count()
```

**Review by charn — APPROVED** [review]
Few comments, but LGTM! :+1: :shipit:

**Pull request merged**